src/DynamoCore/Engine/CodeGeneration/AstBuilder.cs

```diff
@@ -309,8 +224,19 @@ private void CompileToAstNodes(NodeModel node, List<AssociativeNode> resultList,
             if (node.State == ElementState.Error)
                 Log("Error in Node. Not sent for building and compiling");
 
-            if (context == CompilationContext.DeltaExecution || context == CompilationContext.PreviewGraph)
+            if (context == CompilationContext.DeltaExecution)
+            {
+                OnAstNodeBuilding(node.GUID);
+                if (profilingSession != null)
+                {
+                    profilingSession.RegisterNode(node);
+                    resultList.Add(profilingSession.CreatePreCompilationAstNode(node, inputAstNodes));
+                }
+            }
+            else if (context == CompilationContext.PreviewGraph)
+            {
```

Comment:
so what is a PreviewGraph compilation context?

Reply:
I do not know. The code from Sharad called the profiling VMDataBridge code for both, but then explicitly only did anything for DeltaCompute, so I figured it would be clearer to be explicit on the higher level.

Reply:
After discussing today it seems that no-one knows what PreviewGraph is, possibly we need a tech-debt task to figure out.

Reply:
I looked at the preview graph functionality. It is related to node previews and therefore doesn't look like it should be removed any time soon.